server/src/main/java/org/elasticsearch/index/reindex/ReindexRequest.java

```diff
@@ -121,12 +128,84 @@ private boolean routingIsValid() {
         }
     }
 
+    public ReindexRequest setSourceIndices(String... sourceIndices) {
+        if (sourceIndices != null) {
+            this.getSearchRequest().indices(sourceIndices);
+        }
+        return this;
+    }
+
+    public ReindexRequest setSourceDocTypes(String... docTypes) {
+        if (docTypes != null) {
+            this.getSearchRequest().types(docTypes);
+        }
+        return this;
+    }
+
+    public ReindexRequest setSourceBatchSize(int size) {
+        this.getSearchRequest().source().size(size);
+        return this;
+    }
+
+    public ReindexRequest setSourceQuery(QueryBuilder queryBuilder) {
+        if (queryBuilder != null) {
+            this.getSearchRequest().source().query(queryBuilder);
+        }
+        return this;
+    }
+
+    /**
+     * Add a sort against the given field name.
+     *
+     * @param name The name of the field to sort by
+     */
+    public ReindexRequest addSortField(String name, SortOrder order) {
+        this.getSearchRequest().source().sort(name, order);
+        return this;
+    }
+
+    public ReindexRequest setDestIndex(String destIndex) {
```

Comment:
Could you add javadoc for these? They'd be nice because these are part of the public API.

Reply:
I know we weren't good about this in the past but we're *trying* to be better lately.